internal/daemonset/daemonset.go

```diff
@@ -1,198 +1,23 @@
-	}
-	logrus.Infof("Successfully cleaned up daemon set %s", daemonSetName)
-	return nil
-}
-
-// Check if the daemon set exists
-func doesDaemonSetExist(daemonSetName, namespace string) bool {
-	client := clientsholder.GetClientsHolder().K8sClient.AppsV1()
-	_, err := client.DaemonSets(namespace).Get(context.TODO(), daemonSetName, metav1.GetOptions{})
-	if err != nil {
-		logrus.Infof("Error occurred checking for Daemonset to exist: %s", err)
-		return false
-	}
-	return true
-}
-
-//nolint:funlen
-func CreateDaemonSetsTemplate(dsName, namespace, containerName, imageWithVersion string) *v1.DaemonSet {
-	dsAnnotations := make(map[string]string)
-	dsAnnotations["debug.openshift.io/source-container"] = containerName
-	dsAnnotations["openshift.io/scc"] = nodeExporter
-	matchLabels := make(map[string]string)
-	matchLabels["name"] = dsName
-	matchLabels["test-network-function.com/app"] = debug
-
-	var runAsPrivileged = true
-	var zeroInt int64
-	var zeroInt32 int32
-	var preempt = corev1.PreemptLowerPriority
-	var tolerationsSeconds int64 = 300
-	var hostPathType = corev1.HostPathDirectory
-
-	container := corev1.Container{
-		Name:            containerName,
-		Image:           imageWithVersion,
-		ImagePullPolicy: corev1.PullIfNotPresent,
-		SecurityContext: &corev1.SecurityContext{
-			Privileged: &runAsPrivileged,
-			RunAsUser:  &zeroInt,
-		},
-		Stdin:                  true,
-		StdinOnce:              true,
-		TerminationMessagePath: "/dev/termination-log",
-		TTY:                    true,
-		VolumeMounts: []corev1.VolumeMount{
-			{
-				MountPath: "/host",
-				Name:      "host",
-			},
-		},
-	}
-	return &v1.DaemonSet{
-
-		ObjectMeta: metav1.ObjectMeta{
-			Name:        dsName,
-			Namespace:   namespace,
-			Annotations: dsAnnotations,
-		},
-		Spec: v1.DaemonSetSpec{
-			Selector: &metav1.LabelSelector{
-				MatchLabels: matchLabels,
-			},
-			Template: corev1.PodTemplateSpec{
-				ObjectMeta: metav1.ObjectMeta{
-					Labels: matchLabels,
-				},
-				Spec: corev1.PodSpec{
-					Containers:       []corev1.Container{container},
-					PreemptionPolicy: &preempt,
-					Priority:         &zeroInt32,
-					HostNetwork:      true,
-					HostPID:          true,
-					Tolerations: []corev1.Toleration{
-						{
-							Effect:            "NoExecute",
-							Key:               "node.kubernetes.io/not-ready",
-							Operator:          "Exists",
-							TolerationSeconds: &tolerationsSeconds,
-						},
-						{
-							Effect:            "NoExecute",
-							Key:               "node.kubernetes.io/unreachable",
-							Operator:          "Exists",
-							TolerationSeconds: &tolerationsSeconds,
-						},
-						{
-							Effect: "NoSchedule",
-							Key:    "node-role.kubernetes.io/master",
-						},
-						{
-							Effect: "NoSchedule",
-							Key:    "node-role.kubernetes.io/control-plane",
-						},
-					},
-					Volumes: []corev1.Volume{
-						{
-							Name: "host",
-							VolumeSource: corev1.VolumeSource{
```

Comment:
The old code was mounting the host file system in /.
We could add this as an option or add "chroot /host" before every command currently executed using the debug pods

Reply:
For the other places where this is being used, the mounted host file system does not play an important role. So, I would for the time being mount in /.